File: 7/28/4.cpp
#include <string>
#include <vector>

using namespace std;

bool solution(int x) {
    bool answer = true;

    string xs = to_string(x);

    int sum = 0;
    for(int i = 0; i < xs.length(); i++){
        sum += xs[i]-48;
    }

    answer = x%sum == 0;



    return answer;
}

File: 2021/2/5/4.py
import heapq
def solution(scoville, K):
    answer = 0

    pq = []
    for i in range(len(scoville)) :
        heapq.heappush(pq,scoville[i])

    while pq[0] < K :
        answer += 1
        if len(pq) == 1 :
            return -1
        least = heapq.heappop(pq)
        least2 = heapq.heappop(pq)
        heapq.heappush(pq,least+least2*2)
    return answer

